homeassistant/components/apsystems/coordinator.py

```diff
@@ -3,31 +3,24 @@
 from __future__ import annotations
 
 from datetime import timedelta
-import logging
 
 from APsystemsEZ1 import APsystemsEZ1M, ReturnOutputData
 
 from homeassistant.core import HomeAssistant
 from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
 
-_LOGGER = logging.getLogger(__name__)
+from .const import LOGGER
 
 
-class InverterNotAvailable(Exception):
-    """Error used when Device is offline."""
-
-
-class ApSystemsDataCoordinator(DataUpdateCoordinator):
+class ApSystemsDataCoordinator(DataUpdateCoordinator[ReturnOutputData]):
     """Coordinator used for all sensors."""
 
     def __init__(self, hass: HomeAssistant, api: APsystemsEZ1M) -> None:
         """Initialize my coordinator."""
         super().__init__(
             hass,
-            _LOGGER,
-            # Name of the data. For logging purposes.
+            LOGGER,
             name="APSystems Data",
-            # Polling interval. Will only be polled if there are subscribers.
             update_interval=timedelta(seconds=12),
         )
         self.api = api
```

Comment:
The always_update, why is that set here?

Reply:
If I'm not mistaken, HA stops updating the sensor regularly when it throws an error x times in a row. This is set, so it won't stop updating the coordinator.

Reply:
I never heard of that tbf

Reply:
Nah, I've messed up, so thanks for pointing that out.